### ISSUE TITLE
HOTEL BOOKING DEMAND PREDICTION 

### ISSUE THREAD

**Issue by anish3333** [abhisheks008/ML-Crate#665]
### ML-Crate Repository (Proposing new issue)

:red_circle: **Project Title**: Hotel Booking Demand Prediction

:red_circle: **Aim**: Predict hotel booking cancellations and analyze factors affecting hotel bookings.

:red_circle: **Dataset**: https://www.kaggle.com/datasets/jessemostipak/hotel-booking-demand

:red_circle: **Approach**:
1. **Exploratory Data Analysis (EDA)**:
   - Load, explore, and visualize the dataset.

2. **Data Preprocessing**:
   - Handle missing values.
   - Encode categorical variables.
   - Scale numerical features if necessary.

3. **Model Building**:
   - Implement and compare multiple models (at least 3-4 algorithms) such as Logistic Regression, Decision Trees, Random Forest, and Gradient Boosting.

4. **Model Evaluation**:
   - Use cross-validation and appropriate metrics (e.g., accuracy, precision, recall, ROC-AUC) to evaluate model performance.

5. **Hyperparameter Tuning**:
   - Optimize the models using techniques like Grid Search or Random Search.

Please add this project to the repository. Thank you!

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Anish Awasthi
- GitHub Profile Link : https://github.com/anish3333
- Participant ID (If not, then put NA) : NA
- Approach for this Project : Conduct exploratory data analysis (EDA), preprocess the data (handle missing values, encode categorical variables, and scale numerical features), and implement multiple machine learning models. Compare the performance of these models using cross-validation and appropriate metrics, then optimize with hyperparameter tuning to identify the best-performing algorithm.
- What is your participant role? VSOC

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by anish3333** [comment]
I am a contributor of VSOC. Please assign me this issue.

**Comment by abhisheks008** [comment]
Assigned @anish3333 

Implement 6-7 models for this project.

**Comment by abhisheks008** [comment]
> @abhisheks008 If this issue is now available for SSOC-3, please assign it to me. Thanks.
> 
> Full name: Siddhant Tiwari GitHub profile link: https://github.com/siddhant4ds Participant ID: sid4ds (Devfolio), sid4ds (Discord) Participant role: SSOC-3 Contributor Approach:
> 
> 1. Exploratory data analysis
> 2. Implementing variety of models using Scikit-learn, XGBoost, LightGBM, Keras.
> 3. Implement model ensembling through voting, stacking, blending and hill-climbing.
> 4. Create Streamlit web app for inference using the best model.

Finish your assigned issue first, then I'll assign this to you.

**Comment by siddhant4ds** [comment]
@abhisheks008 My previous pull request has been merged. Kindly assign it to me now. Thanks.

**Comment by abhisheks008** [comment]
Assigned @siddhant4ds

**Comment by siddhant4ds** [comment]
@abhisheks008 I have been working on this issue and completed about 80% of my work, but did not want to submit hastily before the deadline.
You can leave this issue assigned to me and I will contribute it after the SSOC deadline. Thanks.
